roachtest: add SQL instance to connection options

herkolategan opened the pull request:
Previously `ConnOption` did not support passing a SQL Instance. This change adds SQL Instance to `ConnOption` and updates the relevant calls to pass the information in order to support specifying a specific SQL instance if required.

A SQL Instance can be viewed as a unique identifier for the SQL Instance that is running on a VM in the event that there are more than one SQL Instance serving the same virtual cluster on that VM.

In the future virtual cluster name and SQL instance could be grouped possibly in struct for convenience.

Epic: None
Release Note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111582)
<!-- Reviewable:end -->

srosenberg reviewed (commented):
At the moment all callers are passing `0`. Is there an imminent use-case where it would be non-zero?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111582#-:-NfvSLZe5eBGhDR0T3pu:b-mmsmby)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @herkolategan and @smg260)

___
*[`pkg/cmd/roachtest/tests/drain.go` line 246 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111582#-NfvStEe6TIIG7_DUsNl:-NfvStEe6TIIG7_DUsNm:bayvb9e) ([raw file](https://github.com/cockroachdb/cockroach/blob/a4c7d470c6a8e22b20c60b757e5d14918da5e3d1/pkg/cmd/roachtest/tests/drain.go#L246)):*
> ```Go
> 	prepareCluster(ctx, t, c, drainWaitDuration, connectionWaitDuration, queryWaitDuration)
> 
> 	pgURL, err := c.ExternalPGUrl(ctx, t.L(), c.Node(nodeToDrain), "" /* tenant */, 0)
> ```

Nit: `/* sqlInstance */ 0` [1]

[1] https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/1881669848/Go+function+argument+patterns#Naive-style%3A-%E2%80%9Crequire-everything%E2%80%9D


<!-- Sent from Reviewable.io -->

renatolabs reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111582#-:-NfwA5Z50jz46D-akcNW:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @herkolategan and @smg260)



<!-- Sent from Reviewable.io -->

herkolategan committed:
roachtest: add SQL instance to connection options

Previously `ConnOption` did not support passing a SQL Instance. This change adds
SQL Instance to `ConnOption` and updates the relevant calls to pass the
information in order to support specifying a specific SQL instance if required.

A SQL Instance can be viewed as a unique identifier for the SQL Instance that is
running on a VM in the event that there are more than one SQL Instance serving
the same virtual cluster on that VM.

In the future virtual cluster name and SQL instance could be grouped possibly in
struct for convenience.

Epic: None
Release Note: None

herkolategan reviewed (commented):
> At the moment all callers are passing `0`. Is there an imminent use-case where it would be non-zero?
At the moment there is one `roachtest` which will use this to start multiple instances on a single VM of the same virtual cluster. But since it's used scarcely, we have discussed using a struct for virtual cluster name and instance to alleviate that.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111582#-:-NgIWEzhEpY32LUL82LM:b7pfazg)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @smg260)

___
*[`pkg/cmd/roachtest/tests/drain.go` line 246 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111582#-NfvStEe6TIIG7_DUsNl:-NgIWATaECXQs2KnAlzD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a4c7d470c6a8e22b20c60b757e5d14918da5e3d1/pkg/cmd/roachtest/tests/drain.go#L246)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Nit: `/* sqlInstance */ 0` [1]

[1] https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/1881669848/Go+function+argument+patterns#Naive-style%3A-%E2%80%9Crequire-everything%E2%80%9D
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

herkolategan commented:
TFTRs!
bors r=renatolabs
